gobblin-runtime/src/main/java/gobblin/runtime/util/FSJobCatalogHelper.java

```diff
@@ -0,0 +1,402 @@
+import gobblin.util.filesystem.PathAlterationObserver;
+import gobblin.runtime.job_catalog.FSJobCatalog.Action;
+
+
+/**
+ * Pretty much the same as SchedulerUtils.java at current stage.
+ * Todo: To completely migrate those parts of code into refactored job launcher & scheduler .
+ */
+
+public class FSJobCatalogHelper {
+  private static final Logger LOGGER = LoggerFactory.getLogger(FSJobCatalogHelper.class);
+
+  // Key used in the metadata of JobSpec.
+  private static final String DESCRIPTION_KEY_IN_JOBSPEC = "description";
+  private static final String VERSION_KEY_IN_JOBSPEC = "version";
+
+  /**
+   * Load job configuration from job configuration files stored in general file system,
+   * located by Path
+   * @param properties Gobblin framework configuration properties
+   *                   A good examplfied option of this is JOB_CONFIG_FILE_GENERAL_PATH_KEY
+   * @return a list of job configurations in the form of {@link JobSpec}
+   */
+  public static List<JobSpec> loadGenericJobConfigs(Properties properties)
+      throws ConfigurationException, IOException {
+    List<JobSpec> jobConfigs = Lists.newArrayList();
+    loadGenericJobConfigsRecursive(jobConfigs, properties, getJobConfigurationFileExtensions(properties),
+        new Path(properties.getProperty(ConfigurationKeys.JOB_CONFIG_FILE_GENERAL_PATH_KEY)));
+    return jobConfigs;
+  }
+
+  /**
+   * Load a given job configuration file from a general file system.
+   *
+   * @param frameworkProps Gobblin framework configuration properties
+   * @param jobConfigRelativePath job configuration file to be loaded
+   * @param jobConfigDirPath The root job configuration directory path
+   * @return a single job configuration in the form of {@link JobSpec}
+   */
+  public static JobSpec loadGenericJobConfig(Properties frameworkProps, Path jobConfigRelativePath,
+      Path jobConfigDirPath)
+      throws ConfigurationException, IOException {
+    // Add the framework configuration properties to the end
+    Properties jobProps = new Properties();
+    jobProps.putAll(frameworkProps);
+
+    // Assemble the complete path in target file system first.
+    // Then load the job configuration properties defined in the job configuration file
+    Path completePath = new Path(jobConfigDirPath, jobConfigRelativePath);
+
+    try (FileSystem fs = completePath.getFileSystem(new Configuration())) {
+      try (InputStream is = fs.open(completePath)) {
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(is, Charsets.UTF_8));
+        jobProps.load(bufferedReader);
+      }
+    }
+
+    if (jobProps.containsKey(ConfigurationKeys.JOB_TEMPLATE_PATH)) {
+      jobProps = (new ResourceBasedTemplate(
+          jobProps.getProperty(ConfigurationKeys.JOB_TEMPLATE_PATH))).getResolvedConfigAsProperties(jobProps);
+    }
+
+    jobProps.setProperty(ConfigurationKeys.JOB_CONFIG_FILE_PATH_KEY, jobConfigRelativePath.toString());
```

Comment:
This attribute is used in `RunOnceJobListener` and `AWSJobConfigurationManager` currently. In the former one, it is used to do some checking job state.